x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/use_columns.tsx

```diff
@@ -468,6 +487,93 @@ export const useGapDurationColumn = () => {
   };
 };
 
+export const useGapStatusColumn = (): TableColumn => {
+  const { euiTheme } = useEuiTheme();
+
+  return useMemo(
+    () => ({
+      field: 'gap_info',
+      name: GAP_STATUS_HEADER,
+      render: (
+        gapInfo:
+          | {
+              total_unfilled_duration_ms: number;
+              total_in_progress_duration_ms: number;
+              total_filled_duration_ms: number;
+              gap_fill_status?: GapFillStatus;
+            }
+          | undefined
+      ) => {
+        const status = gapInfo?.gap_fill_status;
+        if (!gapInfo || !status) return getEmptyTagValue();
+
+        const totalInProgressDurationMs = gapInfo.total_in_progress_duration_ms;
+        const totalUnfilledDurationMs = gapInfo.total_unfilled_duration_ms;
+        const totalFilledDurationMs = gapInfo.total_filled_duration_ms;
+
+        const formatDurationHumanized = (duration: number) => {
+          return duration === 0 ? '0ms' : moment.duration(duration, 'ms').humanize();
+        };
+
+        let color;
+        let label;
+
+        const byStatus: Record<GapFillStatus, { color: string; label: string }> = {
+          [gapFillStatus.IN_PROGRESS]: {
+            color: euiTheme.colors.backgroundBaseWarning,
+            label: GAP_STATUS_IN_PROGRESS_LABEL,
+          },
+          [gapFillStatus.UNFILLED]: {
+            color: euiTheme.colors.backgroundBaseDanger,
+            label: GAP_STATUS_UNFILLED_LABEL,
+          },
+          [gapFillStatus.FILLED]: {
+            color: euiTheme.colors.backgroundBaseSuccess,
+            label: GAP_STATUS_FILLED_LABEL,
+          },
+        };
+
+        const match = byStatus[status];
+
+        if (match) {
+          color = match.color;
+          label = match.label;
+        }
+
+        const tooltip = (
+          <div>
+            <EuiText size="s">
+              {gapStatusTooltipInProgress(formatDurationHumanized(totalInProgressDurationMs ?? 0))}
+            </EuiText>
+            <EuiText size="s">
+              {gapStatusTooltipUnfilled(formatDurationHumanized(totalUnfilledDurationMs ?? 0))}
+            </EuiText>
+            <EuiText size="s">
+              {gapStatusTooltipFilled(formatDurationHumanized(totalFilledDurationMs ?? 0))}
+            </EuiText>
+          </div>
+        );
```

Comment:
nit: Have you considered creating a separate component for the gap fill tooltip? This callback is executed for each row meaning we have N `tooltip` copies.